Projects/Core/Network/Sources/Auth/GoogleAuthService.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  GoogleAuthService.swift
+//  Network
+//
+//  Created by SiJongKim on 6/12/25.
+//
+
+import Foundation
+import GoogleSignIn
+import Alamofire
+import UIKit
+import ServiceInterface
+
+
+public final class GoogleAuthService: GoogleAuthServiceInterface {
+    public init() {}
+    
+    public func signIn() async throws -> (accessToken: String, name: String?, email: String?) {
+        guard let scene = await UIApplication.shared.connectedScenes.first as? UIWindowScene,
```

Comment:
[UIApplication](https://developer.apple.com/documentation/uikit/uiapplication/) 는 메인스레드에서만 접근 가능한데 async 컨텍스트에서 직접 사용이 되고있어요 
```suggestion        
        guard let scene = await MainActor.run { UIApplication.shared.connectedScenes.first } as? UIWindowScene,
      let root = await MainActor.run { scene.windows.first?.rootViewController } else {
```

이런식으로 MainActor로 돌리는것이 좋을것같아요

Reply:
!! 수정하겠습니다 
```swift
let rootViewController = await MainActor.run {
        guard let scene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
              let root = scene.windows.first?.rootViewController else {
            return nil
        }
        return root
    }
    
    guard let root = rootViewController else {
        throw NSError(
            domain: "GoogleAuthService",
            code: -1,
            userInfo: [NSLocalizedDescriptionKey: "RootViewController를 찾을 수 없습니다."]
        )
    }
```
전달 주신 수정 사항에서 살짝 중복 코드 제거하는 방식으로 이렇게 수정하는건 어떠실까요?

Reply:
감사합니다 :) 
전달 주신 수정사항에 대해서 살짝 수정해 이런 방식으로 수정하는 건 어떻게 생각하시나요?
```swift
let rootViewController: UIViewController? = await MainActor.run {
    guard let scene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
          let root = scene.windows.first?.rootViewController else {
        return nil
    }
    return root
}

guard let root = rootViewController else {
    throw NSError(
        domain: "GoogleAuthService",
        code: -1,
        userInfo: [NSLocalizedDescriptionKey: "RootViewController를 찾을 수 없습니다."]
    )
}
```

Reply:
앗 이방법이 더 나은방식인것같네요 제가말한방식은  scene 변수가 첫 번째 MainActor.run 블록 밖에서 정의되지 않아서 두 번째 블록에서 사용할수가 없을거라 이방식이좀더 맞는방법인것같네요!

Reply:
더 근본적인 질문이 있어요..!
제가 코드를 다 본 것은 아니지만 GoogleAuthService에서 rootViewController를 찾는 이유가 뭘까요~?
UI 관련 객체가 아닌 것 같은데 UI에 접근하고 있어서 import UIKit 의존성도 생기고 mainActor 이슈도 생기는 것 같아서 코멘트 남깁니당

Reply:
말씀 해주신 내용을 바탕으로 다시 고민해보니 Service에서 UI에 접근하기 때문에 추가 의존성과 MainActor 이슈가 발생하는 것 같습니다. Service에서는 비즈니스 로직/API통신만 담당 하도록 그 책임을 명확하게 할 필요가 있을 것 같아요.

LoginWithGoogleUseCaseInterface를 구성해서 UI에 접근하는 로직을 별도 구성하고 현재 Store에서 추가적으로 담당하고 있는 handleGoogleSignIn로 해당 UseCase에서 구현하여 Store가 불필요한 로직을 가지고 있지 않도록 하는 방식으로 시도해보겠습니다.